### PR TITLE
define tls.Config explicitly instead redefine default TLS config options 

### PR DESCRIPTION
**Pull request by asmyasnikov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

The SDK's default tls.Config contains root CA. User may appends custom certificates to defaults TLS config, redefine min TLS version and change InsecureSkipVerify flag. 
Also internal config field secure is an ambigious with TLS config. Always true secure flag indirectly redefines tls config with default TLS, false secure flag skips TLS config to nil.
PR's contains changes for using user-defined tls.Config explicitly and deprecates other options about TLS config.

CHANGELOG:
* Added `WithTLSConfig` option for redefine TLS config
* Marked as deprecated options `WithSecure`, `WithInSecure`, `WithMinTLSVersion`, `WithTLSSInsecureSkipVerify`, `WithCertificate`, `WithCertificatesFromFile, `WithCertificatesFromPem`
* Added `sugar.LoadCertificatesFromFile` and `sugar.LoadCertificatesFromPem` helpers

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #217 

## What is the new behavior?

-
-
-

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -218,12 +244,18 @@ func WithRequestsType(requestsType string) Option {
 	}
 }
 
+// WithMinTLSVersion applies minimum TLS version that is acceptable.
+//
+// Deprecated: use explicit WithTLSConfig option instead
 func WithMinTLSVersion(minVersion uint16) Option {
 	return func(c *Config) {
 		c.tlsConfig.MinVersion = minVersion
```
Comment (rekby) [inline config/config.go:252]: Here and in similar places - may be check c.tlsConfig != nil or force WithSecure(true) for prevent nil panic on config apply?

File: sugar/certificates.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xerrors"
+)
+
+func LoadCertificatesFromFile(caFile string) ([]*x509.Certificate, error) {
+	if len(caFile) > 0 && caFile[0] == '~' {
```
Comment (rekby) [inline sugar/certificates.go:14]: In common libraries doesn't need shell expand.
I suggent to use caFile as path "as is"

File: config/config.go (hunk 1 above)
Comment (asmyasnikov) [reply config/config.go:239]: You right, ack

File: sugar/certificates.go (hunk 2 above)
Comment (asmyasnikov) [reply sugar/certificates.go:15]: ack

**Pull request merged**